config/configauth/authenticator.go

```diff
@@ -17,53 +17,48 @@ package configauth
 import (
 	"context"
 	"errors"
-	"io"
 
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
+
+	"go.opentelemetry.io/collector/component"
 )
 
 var (
-	errNoOIDCProvided   = errors.New("no OIDC information provided")
 	errMetadataNotFound = errors.New("no request metadata found")
-	defaultAttribute    = "authorization"
 )
 
-// Authenticator will authenticate the incoming request/RPC
+// Authenticator is an Extension that can be used as an authenticator for the configauth.Authentication option.
+// Authenticators are then included as part of OpenTelemetry Collector builds and can be referenced by their
+// names from the Authentication configuration. Each Authenticator is free to define its own behavior and configuration options,
+// but note that the expectations that come as part of Extensions exist here as well. For instance, multiple instances of the same
+// authenticator should be possible to exist under different names.
 type Authenticator interface {
-	io.Closer
+	component.Extension
 
 	// Authenticate checks whether the given context contains valid auth data. Successfully authenticated calls will always return a nil error and a context with the auth data.
+	// Implementations should add the derived subject (user, principal) to a new context built based on the given context under the key configauth.SubjectKey.
```

Comment:
Also, I believe we discussed we will have a different place for storing auth information instead of Context so that it survives passing through the pipeline, but that is a separate topic.
I guess I do not understand why we store this information in the context at all for now since OIDC authenticator does not seem to need to pass any info in the pipeline, does it?

Reply:
>Is every authenticator expected to have a concept of user and a concept of principal?

No. In the different specs around auth, there are different names for pretty much the same thing: Subject, User, Principal, which is why I'm listing them here as synonyms. I should perhaps make it clear here that they are all the same thing.

>I would assume authenticator is free to put whatever information they want in the context.

Agree, but I foresee three fields to be added to the pdata structures: subject, raw, membership. Membership is for the case where the authenticator can infer to which groups the subject belongs to.

Reply:
>I guess I do not understand why we store this information in the context at all for now since OIDC authenticator does not seem to need to pass any info in the pipeline, does it?

Good point, this is basically what I had before in the old auth scheme, but it's probably better to remove this.

Reply:
While removing the context operations for the authenticator, in favor of the future change of adding them as part of the resource, I removed this code comment as well.